ocp_resources/node_network_configuration_policy.py

```diff
@@ -374,20 +374,25 @@ def wait_for_status_success(self):
         failed_condition_reason = self.Conditions.Reason.FAILED_TO_CONFIGURE
         no_match_node_condition_reason = self.Conditions.Reason.NO_MATCHING_NODE
 
-        # if we get here too fast there are no conditions, we need to wait.
-        self.wait_for_configuration_conditions_unknown_or_progressing()
+        def _get_available_status_condition() -> dict[str, str]:
+            conditions = [
+                condition
+                for condition in self.instance.get("status", {}).get("conditions", [])
+                if condition and condition["type"] == self.Conditions.Type.AVAILABLE
+            ]
+            return conditions[0] if conditions else {}
 
-        samples = TimeoutSampler(wait_timeout=self.success_timeout, sleep=1, func=lambda: self.status)
         try:
-            for sample in samples:
-                if sample == self.Conditions.Reason.SUCCESSFULLY_CONFIGURED:
+            for sample in TimeoutSampler(
+                wait_timeout=self.success_timeout, sleep=5, func=_get_available_status_condition
+            ):
+                if sample.get("status") == self.Condition.Status.TRUE:
```

Comment:
Maybe it's better to do here
```
if not sample:
    continue
```

Reply:
Since I am returning {} if no condition is found, I don't think I need the check for sample. if sample.get("status") would be None by default.

Reply:
Since you can tell that sample may be empty you can skip all other calls.
There is no point to check 3 different things while you know that sample is empty.

Yes, the code will not fail, but it's not optimized.